### ISSUE TITLE
[BUG]404 page not found on

### ISSUE THREAD

**Issue by josch** [nanu-c/axolotl#954]
### Description

[Description of the issue]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

#### Versions

Please provide the Version as written in `Settings->About Axolotl`

#### Device
What device or OS are you using?

#### Link to Debug Log
Please provide a link to debug from ~/.cache/upstart/application-click-textsecure.nanuc_textsecure*.log
Be careful it contains sensible data

**Comment by nuehm-arno** [comment]
Hey josch,
logs for Mobian can be found like this:
`journalctl --since "2022-03-01 10:00:00" | grep -E 'axolotl|textsecure' > axolotl_logs`
with your actual timestamp of interest.
And another question: You said, your OS is Debian usntable on a laptop - is this an x86_64 machine? If so, Axolotl won't work with the deb file installation, because it's built for arm64 machines only.
Cheers

**Comment by josch** [comment]
No, my laptop is arm64 and not x86_64 -- otherwise the `apt install ./axolotl_1.6.0-1_arm64.deb` step wouldn't have worked. :slightly_smiling_face:  

```
$ dpkg --print-architecture
arm64
```

Since I filed this bug 9 days ago, I rebooted the machine and now the 404 is gone. The reason is the environment variables exported in `/etc/profile.d/axolotl.sh`:

```
# Set environment variables for Axolotl
export AXOLOTL_WEB_DIR=/usr/share/axolotl/axolotl-web/dist
export AXOLOTL_GUI_DIR=/usr/share/axolotl/
```

Right after installing `axolotl_1.6.0-1_arm64.deb`, those environment variables are not set yet for the currently running system. It needs a reboot or a re-login or a manual sourcing of `/etc/profile.d/axolotl.sh` for these variables to take effect.

So this is not a bug in axolotl itself but a bug in the packaging. I'll leave it up to you close this issue or act upon it, for example by making axolotl not need these environment variables.

Now that I no longer get a 404, my next issue is #943 but I already subscribed to that one. :slightly_smiling_face:

**Comment by nuehm-arno** [comment]
Thanks for the information. You're the first to report the missing sourcing. Normally, the Debian `postinst` script triggers `/etc/profile.d/axolotl.sh` to set the variables, but in your case, it didn't. That is a new one and I don't understand why. I'll do some research!

**Comment by josch** [comment]
The answer is easy. The postinst runs this:

    bash -c "source /etc/profile.d/axolotl.sh"
    exit 0

So you spawn a bash process and that sources the axolotl.sh. From that point on, that bash process will have all the variables in axolotl.sh set. But then the bash process exits. The shell that spawned bash knows nothing about the variables of the bash process that just exited. Then the postinst shell script exits.

In short: the variables you set in a maintainer script have no influence over the variables set in a user shell.
